tests/kafkatest/services/kafka/kafka.py

```diff
@@ -165,6 +165,19 @@ def start(self, add_principals=""):
         self.start_minikdc(add_principals)
         Service.start(self)
 
+        self.logger.info("Waiting for brokers to register at ZK")
+
+        cmd = self.path.script("zookeeper-shell.sh", self.zk) + " " + self.zk_connect_setting() + " ls /brokers/ids"
+
+        broker_ids = ""
+        for node in self.nodes:
+            broker_ids += "%s, " % self.idx(node)
+        broker_ids = "[" + broker_ids[:-2] + "]"
+
+        with self.zk.account.monitor_log(KafkaService.STDOUT_STDERR_CAPTURE) as monitor:
+            self.zk.account.ssh(cmd)
```

Comment:
definition of `cmd` seems weirdly separated from execution here. not critical, but moving it into the `with` makes things clearer.